app/facades/canvas_facade.rb

```diff
@@ -0,0 +1,265 @@
+      &.assignment_override_students.any? { |studentError| studentError&.type != "taken" }
+    )
+      return createOverrideResponse
+    end
+
+    currOverride = get_existing_student_override(courseId, studentId, assignmentId)
+    if (currOverride == nil)
+      raise NotFoundError.new('could not find student\'s override')
+    end
+    if (currOverride.student_ids.length == 1)
+      return update_assignment_override(
+        courseId,
+        assignmentId,
+        currOverride.id,
+        currOverride.student_ids,
+        overrideTitle,
+        get_current_formatted_time(),
+        newDueDate,
+        newDueDate
+      )
+    end
+    remove_student_from_override(courseId, currOverride, studentId)
+    create_assignment_override(
+      courseId,
+      assignmentId,
+      [studentId],overrideTitle,
+      newDueDate,
+      get_current_formatted_time(),
+      newDueDate
+    )
+  end
+
+  private
+
+  ##
+  # Gets the existing override for a student.
+  #
+  # @param  [Integer] courseId the courseId to check for an existing override.
+  # @param  [Integer] studentId the student to check for an existing override for.
+  # @param  [Integer] assignmentId the assignmnet to check for an existing override for.
+  # @return [OpenStruct|nil] the override if it is found or nil if not.
+  # @throws [FailedPipelineError] if the existing overrides response body could not be parsed.
+  def get_existing_student_override(courseId, studentId, assignmentId)
+    begin
+      allAssignmentOverrides = JSON.parse(
+        get_assignment_overrides(courseId, assignmentId).body,
+        object_class: OpenStruct
+      )
+    rescue JSON::ParserError
+      raise FailedPipelineError.new(
+        'Update Student Extension',
+        'Get Existing Student Override',
+        'Parse Canvas Response'
+      )
+    end
+
+    allAssignmentOverrides.each do |override|
+      if (override&.student_ids.include?(studentId))
+        return override
+      end
+    end
+    return nil
+  end
+
+  ##
+  # Gets the current time as formatted for Canvas's version of iso8601.
+  #
+  # @return [String] the current time that Canvas likes.
+  def get_current_formatted_time
+    currDateTimeUnformatted = DateTime.now().iso8601
+    # This is some weird format of iso8601 standard that Canvas likes... Don't ask me.
+    /[0-9]{4}\-[0-9]{2}\-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}/.match(currDateTimeUnformatted)[0] + 'Z'
+  end
+
+  ##
+  # Removes a student from an existing override.
+  #
+  # @param   [Integer]    courseId the courseId to remove the student from the override of.
+  # @param   [OpenStruct] override the existing override to remove the student from.
+  # @param   [Integer]    studentId the id of the student to remove from the override.
+  # @return  [Faraday::Response] the new override if successful.
+  # @raises  [FailedPipelineError] if the student could not be removed from the override.
+  def remove_student_from_override(courseId, override, studentId)
```

Comment:
This function never actually makes an api call, and from what I can tell neither does where it's called (line 182). If I'm understanding it right, it should call update_assignment_override so as to actually remove the student on the canvas side.

Reply:
Yep!  It calls `update_assignment_override` on line 245

Reply:
Ooops sorry lol, not sure how I missed that